integration/experiment/util.py

```diff
@@ -166,3 +166,43 @@ def merge_const_config(new_config, old_config_path=None):
     jsonschema.validate(combined_config, schema=schema)
 
     return combined_config
+
+def energy_efficient_frequency(df, freq_col_name, energy_col_name, energy_margin):
+    """
+    Find the frequency that provides the minimum package energy consumption
+    within the dataframe provided.
+    """
+    df_mean = df.groupby(freq_col_name)[energy_col_name].mean()
+    energy_efficient_frequency = df_mean.idxmin()
+    min_energy = df_mean[energy_efficient_frequency];
+
+    if len(df_mean) > 1:
+        if energy_margin != 0.0:
+            sys.stderr.write('Found Fe = {} with energy = {}.  Searching for alternate '
+                             'based on an energy margin of {}\n'.format(energy_efficient_frequency,min_energy, energy_margin))
+
+            # Grab all energy readings associated with frequencies that a 1Hz below Fe
+            # TODO: Consider iloc instead and just grab all idx prior to Fe
+            df_mean = df_mean.loc[:energy_efficient_frequency - 1]
+
+            # Find any energy reading that is within 5% of Fe's energy while having a lower frequency
+            energy_in_margin = [e for e in df_mean if (e - min_energy) / e < energy_margin]
+
+            # If any values are within margin
+            if energy_in_margin:
+                min_energy = max(energy_in_margin);
+                # Store the associated frequency
+                energy_efficient_frequency = df_mean[df_mean == min_energy].index[0];
```

Comment:
```suggestion
                energy_efficient_frequency = df_mean.loc[df_mean == min_energy].index[0];
```
Or  idxmin() might work.